src/components/PersonList.tsx

```diff
@@ -0,0 +1,128 @@
+import { useEffect, useState } from 'react';
+import { Loader } from './Loader';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import classNames from 'classnames';
+import { NavLink, useParams } from 'react-router-dom';
+
+export const PersonList = () => {
+  const [peopleFromApi, setPeopleFromApi] = useState<Person[]>([]);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const { slug } = useParams();
+
+  const selectedPerson = slug ? slug : '';
+
+  useEffect(() => {
+    setLoading(true);
+    getPeople()
+      .then(people => {
+        const findParents = people.map(person => {
+          return {
+            ...person,
+            mother: people.find(child => child.name === person.motherName),
+            father: people.find(child => child.name === person.fatherName),
+          };
+        });
+
+        setPeopleFromApi(findParents);
+      })
+      .catch(() => {
+        setError(true);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  return (
+    <main className="section">
+      <div className="container">
+        <h1 className="title">People Page</h1>
+
+        <div className="block">
+          <div className="box table-container">
+            {error && (
+              <p data-cy="peopleLoadingError" className="has-text-danger">
+                Something went wrong
+              </p>
+            )}
+
+            {!loading && !peopleFromApi.length && (
+              <p data-cy="noPeopleMessage">There are no people on the server</p>
+            )}
+            {loading ? (
+              <Loader />
+            ) : (
+              <table
```

Comment:
if error occurs there shouldn't be message about no people on server and table should appear only when there is some data in it
![image](https://github.com/user-attachments/assets/1e0c6e76-d2fc-46a4-9b93-35eb116c67fd)